src/dataset-builder/DatasetBuilder.ts

```diff
@@ -709,12 +709,12 @@ export const DatasetBuilderView: React.FC<DatasetBuilderProps> = (props) => {
                     })
                   : div(['No Dataset Builder Settings Found']);
               case 'domain-criteria-selector':
-                return h(DomainCriteriaSelector, { state: datasetBuilderState, onStateChange });
+                return h(DomainCriteriaSelector, { state: datasetBuilderState, onStateChange, datasetId });
               case 'concept-set-creator':
                 return datasetDetails.state.snapshotBuilderSettings
                   ? h(ConceptSetCreator, {
                       onStateChange,
-                      snapshotBuilderSettings: datasetDetails.state.snapshotBuilderSettings,
+                      dataset: datasetDetails.state,
```

Comment:
Here's another case where our names seem to be disagreeing. We have the names `dataset` `DatasetModel` and `datasetDetails` that all seem to refer to the same thing. If they do, they should probably all have the same name instead of different ones. It's probably OK to have `DatasetModel` and then assign it to `dataset` since `Model` doesn't really mean much, but mixing in `details` makes it confusing IMO.